opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -386,9 +387,15 @@ namespace Opm {
         catch ( std::runtime_error& e )
         {
             const std::string msg = "A zero well potential is returned for output purposes. ";
-            OpmLog::warning("WELL_POTENTIAL_CALCULATION_FAILED", msg);
+            local_deferredLogger.warning("WELL_POTENTIAL_CALCULATION_FAILED", msg);
         }
         previous_well_state_ = well_state_;
+
+        Opm::DeferredLogger global_deferredLogger = gatherDeferredLogger(local_deferredLogger);
+        if (terminal_output_) {
```

Comment:
If you change `terminal_output_` as suggested elsewhere this check will be true on all processes. And by design of `gatherDeferredLogger` (which uses MPI_Allgatherv) all processes will have all messages and log them. Hence multiples of all messages.

Maybe gatherDeferredLogger should be refactored to gather the messages only to one process?

Reply:
see other comments